runelite-client/src/main/java/net/runelite/client/plugins/skillcalculator/SkillCalculator.java

```diff
@@ -121,8 +123,9 @@ void openCalculator(CalculatorType calculatorType)
 		// Update internal skill/XP values.
 		currentXP = client.getSkillExperience(calculatorType.getSkill());
 		currentLevel = Experience.getLevelForXp(currentXP);
-		targetLevel = enforceSkillBounds(currentLevel + 1);
-		targetXP = Experience.getXpForLevel(targetLevel);
+		VarPlayer endGoal = endGoalVarpForSkill(calculatorType.getSkill());
+		targetXP = endGoal != null ? client.getVar(endGoal) : 0;
```

Comment:
This will always set the target to 0 xp/level 1 if there is no goal.

> One thing to note is that I did not do any testing on this

Can you test this?

Reply:
I did do a manual `targetXP = 0` before committing and it looks like it always just sets the target level to `currentLevel + 1` but I don't have any 99's so I can't test that it works always. Guess I can just throw a case in there for using the current method (i.e., `currentLevel + 1`) just to be sure. What do you think?

Reply:
If you have 99s or have no goal set the vars are just 0. I don't see why it would set the target level to currentLevel + 1 if you are removing the code which does that? It looks to me like if the target was `0` it would set `targetXP=0` and `targetLevel=1`, which would be wrong.

Reply:
Yeah I was confused when I saw that too, but regardless I've gone ahead and made the necessary changes. I made sure to go in-game and test all options related to end goals and now it looks like everything works as intended. Hopefully you'll agree!